pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -23,13 +23,14 @@ func NewDualWriterMode2(legacy LegacyStorage, storage Storage) *DualWriterMode2
 
 // Create overrides the default behavior of the DualWriter and writes to LegacyStorage and Storage.
 func (d *DualWriterMode2) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	legacy, ok := d.legacy.(rest.Creater)
+	legacy, ok := d.Legacy.(rest.Creater)
 	if !ok {
 		return nil, fmt.Errorf("legacy storage rest.Creater is missing")
 	}
 
 	created, err := legacy.Create(ctx, obj, createValidation, options)
 	if err != nil {
+		klog.FromContext(ctx).Error(err, "unable to create object in legacy storage", "mode", 2)
```

Comment:
Why do we use the `klog` instead of the `grafana/pkg/infra/log` one?

Reply:
There's a discussion above about that, we will talk about it in the next council :) for now it was decided we would use `klog`, in order to block this PR.